### PR TITLE
Fix `tsh play` `--skip-idle-time` not working correctly

### PR DESCRIPTION
**Pull request by gabrielcorado**
Providing the `--skip-idle-time` flag to `tsh play` wasn't working correctly given two issues:
1. The `cfg.SkipIdleTime` wasn't being passed to the `Player` `skipIdleTime` struct property. This property is used by `applyDelay` to decide whether to use the skip idle logic.
2. On the skip idle logic at `applyDelay`, the value used to compare with the event's time difference was too large (given that we were multiplying it by `time.Millesecond` before the comparison). Given this, the value would most likely not be used.

Changelog: Fix `tsh play` not skipping idle time when `--skip-idle-time` was provided.

### CODE REVIEW

File: lib/player/player_test.go [hunk 1]
```diff
@@ -321,6 +322,34 @@ func TestUseDatabaseTranslator(t *testing.T) {
...
+		// events before the timer is set.
+		require.EventuallyWithT(t, func(t *assert.CollectT) {
+			clk.Advance(time.Duration(player.MaxIdleTimeMilliseconds) * time.Millisecond)
+			select {
+			case evt := <-p.C():
+				assert.Equal(t, int64(i), evt.GetIndex())
+			default:
+				assert.Fail(t, "expected to receive event after short period, but got nothing")
+			}
+		}, time.Second, 100*time.Millisecond)
```
Comment (zmb3) [inline lib/player/player_test.go:349]: 1 second rarely seems enough in our CI, should we make it 3?

**Review by zmb3 — APPROVED** [review]
Thank you for catching this. I must have messed up a merge conflict or something 🤷‍♂️

File: lib/player/player.go [hunk 2]
```diff
@@ -81,7 +81,12 @@ type Player struct {
 	translator sessionPrintTranslator
 }
 
-const normalPlayback = math.MinInt64
+const (
+	normalPlayback = math.MinInt64
+	// MaxIdleTimeMilliseconds defines the max idle time when skipping idle
+	// periods on the recording.
+	MaxIdleTimeMilliseconds = 500.0
```
Comment (espadolini) [inline lib/player/player.go:88]: Can we convert this and the calculations into `time.Duration` ones instead? `Player.speed` being an `atomic.Value` is also pretty awful.

**Review by espadolini — APPROVED** [review]
Why is `Player.speed` an `atomic.Value`?

File: lib/player/player.go (hunk 2 above)
Comment (gabrielcorado) [reply lib/player/player.go]: I've converted the timing values into `time.Duration`.

About the `speed` attribute, it needs lock protection, given that users might change the playback speed at any time during the session replay (through the `SetSpeed` function). I'm unsure if your comment relates to the necessity of the atomic value or the usage of `atomic.Value` specific. I will go ahead and merge this PR so we can have the fix released, and I can open a follow-up on this refactor (if we require it).

Comment (espadolini) [reply lib/player/player.go]: The speed value being atomic makes sense, using an `atomic.Value` is really awful, forcing allocations and type assertions every time we interact with it.

If we ever refactor this we should just use an `atomic.Uint64` and `math.Float64bits`/`Float64frombits` instead - it's not worth doing a change specifically for this, however.

File: lib/player/player.go [hunk 3]
```diff
@@ -135,18 +140,19 @@ func New(cfg *Config) (*Player, error) {
...
+		emit:         make(chan events.AuditEvent, 1024),
+		playPause:    make(chan chan struct{}, 1),
+		wake:         make(chan time.Duration),
+		done:         make(chan struct{}),
+		skipIdleTime: cfg.SkipIdleTime,
 	}
 
 	p.speed.Store(float64(defaultPlaybackSpeed))
-	p.advanceTo.Store(normalPlayback)
+	p.advanceTo.Store(normalPlayback.Milliseconds())
```
Comment (espadolini) [inline lib/player/player.go:155]: There's no need to round to milliseconds every time; `time.Duration` is already an int64 storing nanoseconds, you can just do this instead:

```suggestion
	p.advanceTo.Store(int64(normalPlayback))
```

File: lib/player/player.go [hunk 4]
```diff
@@ -184,7 +190,7 @@ func (p *Player) stream() {
 	defer cancel()
 
 	eventsC, errC := p.streamer.StreamSessionEvents(ctx, p.sessionID, 0)
-	lastDelay := int64(0)
+	lastDelay := time.Duration(0)
```
Comment (espadolini) [inline lib/player/player.go:193]:
```suggestion
	var lastDelay time.Duration
```

File: lib/player/player.go [hunk 5]
```diff
@@ -216,20 +222,20 @@ func (p *Player) stream() {
 
 			currentDelay := getDelay(evt)
 			if currentDelay > 0 && currentDelay >= lastDelay {
-				switch adv := p.advanceTo.Load(); {
+				switch adv := time.Duration(p.advanceTo.Load()) * time.Millisecond; {
```
Comment (espadolini) [inline lib/player/player.go:225]:
```suggestion
				switch adv := time.Duration(p.advanceTo.Load()); {
```

File: lib/player/player.go [hunk 6]
```diff
@@ -216,20 +222,20 @@ func (p *Player) stream() {
...
 				case adv >= currentDelay:
 					// no timing delay necessary, we are fast forwarding
 					break
 				case adv < 0 && adv != normalPlayback:
 					// any negative value other than normalPlayback means
 					// we rewind (by restarting the stream and seeking forward
 					// to the rewind point)
-					p.advanceTo.Store(adv * -1)
+					p.advanceTo.Store(adv.Milliseconds() * -1)
```
Comment (espadolini) [inline lib/player/player.go:233]:
```suggestion
					p.advanceTo.Store(int64(adv) * -1)
```

File: lib/player/player.go [hunk 7]
```diff
@@ -216,20 +222,20 @@ func (p *Player) stream() {
...
 					go p.stream()
 					return
 				default:
 					if adv != normalPlayback {
-						p.advanceTo.Store(normalPlayback)
+						p.advanceTo.Store(normalPlayback.Milliseconds())
```
Comment (espadolini) [inline lib/player/player.go:238]:
```suggestion
						p.advanceTo.Store(int64(normalPlayback))
```

File: lib/player/player.go [hunk 8]
```diff
@@ -257,7 +263,7 @@ func (p *Player) stream() {
 			//
 			// TODO: consider a select with a timeout to detect blocked readers?
 			p.emit <- evt
-			p.lastPlayed.Store(currentDelay)
+			p.lastPlayed.Store(currentDelay.Milliseconds())
```
Comment (espadolini) [inline lib/player/player.go:266]:
```suggestion
			p.lastPlayed.Store(int64(currentDelay))
```

**Review by espadolini — APPROVED** [review]

**Pull request merged**